### DISCUSSION TITLE
Problem with new v2.9.0 update: decompressing block: incorrect gzip header check

### DISCUSSION THREAD

**Discussion by SamCT** [GMOD/jbrowse-components#4120] [General]
Hello,

I've got two version of Jbrowse. I just updated one to 2.9.0, the other is 2.8.0.

The update to 2.9.0 has resulted in my .bam files having this error: Error: problem decompressing block: incorrect gzip header check
 when attempting to load. These same bam files, no change, are able to be loaded correctly in my version 2.8.0 of jbrowse2.

I am wondering what is an easy fix. Seems odd. I reran samtools index for the bam files, but that didn't seem to do anything. Curious if there are any suggestions from the Jbrowse team.

Thanks,
Sam

**Comment by cmdcolin** [comment]
there shouldn't be any differences with bam behavior between 2.8.0 and 2.9.0. I can maybe...maybe think of some sort of transitive effects that could affect bam in v2.9.0 that are a little different from v2.8.0, but it seems to me unlikely to cause this

can you ensure that it is the exact same file and same region being loaded between both versions and clear cache in your web browser also?

**Reply by SamCT** [reply]
I just confirmed it's the same files and same region. There is basically no difference between track details. Cleared cache, same issue. Working fine on 2.8.0 but not 2.9.0. 

Sam

**Reply by cmdcolin** [reply]
wow that is very interesting. if you are able to email me the BAM and BAI and FASTA probably, it might be the easiest way for me to reproduce (colin.diesh@gmail.com). if you can indicate which browser you are using it could also help.

**Reply by cmdcolin** [reply]
if your instance is online can check that out directly also

**Reply by SamCT** [reply]
Just sent you an email.

**Comment by SamCT** [comment]
The solution was to remove the mime_magic module in Apache. This seems to have fixed the issue and I have not yet seen any downstream effects from removing this module. Thank you again Colin for your help with this issue.

**Comment by cmdcolin** [comment]
re-uploading conversation from email:

thanks for sending me this info. I downloaded the BAM file and it works ok locally. Do you have any info on the server you are using? I think it could be a server side bug or a server side behavior that could be fixed. Specifically, one of the things that is noticeable is that your server is serving the BAM data with "Content-Encoding: gzip". This is NOT what JBrowse wants to see.

This issue has discussed in the past in various places, it's a tricky bug, but it affects BAM files, VCF files, and more and causes lots of mysterious bugs, but it is a server side behavior

1. http://gmod.org/wiki/JBrowse_Configuration_Guide#Apache_Configuration_Note
2. https://jbrowse.org/docs/faq_troubleshooting.html#what-is-the-error-invalid-bgzf-header-on-my-vcf-files
3. https://github.com/GMOD/jbrowse/issues/825 (many other jbrowse 1 issue threads sort of like this)

Oftentimes we suggest to make it so your server does not add this header. If you are using Apache, we suggest turning off the mime_magic module.

My thought is maybe the v2.8.0 and v2.9.0 are using different servers and maybe the server that is hosting v2.8.0 is not adding Content-Encoding:gzip, which causes it to work?

Here is how I find the headers: i go into devtools, then network log, then look at the response headers on the BAM file request

![Screenshot from 2023-12-10 16-41-54(1)](https://github.com/GMOD/jbrowse-components/assets/6511937/0e550a87-3936-4f1e-ae1b-e7cbe9711049)
